File: frontend/src/modules/labour/services/workerService.ts
import { apiGet, apiPost, apiPut } from '@/services/apiClient';
import type { PageResponse } from '@/shared/types/api';
import type { WorkerRequest, WorkerResponse, WorkerSearchParams } from '../types';

/** Backend: labour/controller/WorkerController.java */
export const workerService = {
  search: (params: WorkerSearchParams) =>
    apiGet<PageResponse<WorkerResponse>>('/v1/workers', { params }),

  listActive: () => apiGet<WorkerResponse[]>('/v1/workers/active'),

  get: (id: number) => apiGet<WorkerResponse>(`/v1/workers/${id}`),

  create: (body: WorkerRequest) => apiPost<WorkerResponse>('/v1/workers', body),

  update: (id: number, body: WorkerRequest) => apiPut<WorkerResponse>(`/v1/workers/${id}`, body),

  deactivate: (id: number) => apiPost<void>(`/v1/workers/${id}/deactivate`),
  activate: (id: number) => apiPost<void>(`/v1/workers/${id}/activate`),
};

File: backend/src/main/java/com/hardware/erp/labour/service/WorkerService.java
package com.hardware.erp.labour.service;

import com.hardware.erp.common.dto.PageResponse;
import com.hardware.erp.labour.dto.WorkerRequest;
import com.hardware.erp.labour.dto.WorkerResponse;
import com.hardware.erp.labour.entity.WorkerStatus;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface WorkerService {

    WorkerResponse create(WorkerRequest request);

    WorkerResponse update(Long id, WorkerRequest request);

    WorkerResponse get(Long id);

    PageResponse<WorkerResponse> search(String search, WorkerStatus status, Pageable pageable);

    /** Active workers only, for attendance-marking and payment pickers. */
    List<WorkerResponse> listActive();

    void deactivate(Long id);

    void activate(Long id);
}
